@here/harp-mapview/lib/Tile.ts

```diff
@@ -2072,20 +2061,67 @@ export class Tile implements CachedResource {
      * Creates and add a background plane for the tile.
      */
     private addGroundPlane() {
-        // Add a ground plane to the tile.
-        const planeSize = new THREE.Vector3();
-        this.boundingBox.getSize(planeSize);
-        const groundPlane = this.createPlane(
-            planeSize.x,
-            planeSize.y,
-            this.center,
+        const color =
             this.dataSource.tileBackgroundColor === undefined
                 ? this.mapView.clearColor
-                : this.dataSource.tileBackgroundColor,
-            this.dataSource.tileBackgroundIsVisible
-        );
+                : this.dataSource.tileBackgroundColor;
+
+        if (this.projection.type === ProjectionType.Spherical) {
```

Comment:
I agree, and honestly I don't think this "background tile" makes any sense. I just added the support for spherical surfaces to the "background tile" just to make to keep the current change as small as possible.
What I would like to see is an internal BackgroundTileDataSource that the mapview initializes and uses it to provide background to all the data sources. For globe, for example, there's no need to have tessellated quads! a big sphere centered at 0 would be enough and it will also fix the holes at the poles.

Reply:
yes, there is a createPlane, there should also be a createCurvedPlace... or similar

Reply:
I don't think that Tile is the correct place for createPlane/createCurvedPlane. Doing it there means that potentially every data source will get a replacement tile, creating a high number of similar geometries. Also, Tile is not accessible for a worker thread, and usually scaling geometries to the surface is something that will be done at decoding time.
I'm trying to keep this change as small as possible, so I kept the current 'background tile' working with sphere, but I would propose to rework the way we do background tiles completely in a follow up change.